src/kml/ext_data.rs

```diff
@@ -1,11 +1,64 @@
 use std::io::Write;
 
-use courageous_format::Record;
+use courageous_format::{
+    Alarm, Classification, CoordENU, DetectionRecord, Location, Position3d, TrackingRecord,
+};
 use quick_xml::{
     events::{BytesCData, BytesText, Event},
     Writer,
 };
 
+pub struct Record {
```

Comment:
Is this actually needed?

Reply:
I'm not an expert on rust, but write_extended_data requires a record type as an argument. As of now, there are two types which require such method: TrackingRecord and DetectionRecord. The only two solutions I can think of are creating 2 variants of the function for each type or transforming them into a common type.

TrackingRecord requires write_extended_data for tracks that don't use Position2d or Position3d, as those are required for gx track to work, which is used for write_track_data or write_track_extended_data (I'm not sure of the name).

This means that it is inevitable that both record types use write_extended_data. Another solution could be to expand the functionality of write_track_extended_data but that seems too cumbersome for the moment.

Reply:
`write_extended_data` will get more and more different for each tracking/detection record variant as time goes on, so I think it makes sense to just have a function for each.

Reply:
> Maybe for a later PR we could even make a function that takes in kml schema member names and their values and just writes that, adding all the SimpleData/SchemaData/schemaUrl boilerplate as needed.

Yes, I agree that would solve most of this problems.
Either way, I've made both a conversion to Record as well as split the write_extended_data function on an earlier node (though I haven't fused write_tracking_extended_data with write_track_extended_data) so we have both options readily available.